src/OpalCompiler-Core/CompilationContext.class.st

```diff
@@ -78,32 +113,202 @@ CompilationContext class >> defaultTransformationPlugins: anObject [
+]
+
+{ #category : #'options - settings API' }
+CompilationContext class >> optionParseErrors: aBoolean [
+	^ self writeDefaultOption: #optionParseErrors value: aBoolean
+]
+
+{ #category : #'options - settings API' }
+CompilationContext class >> optionReadOnlyLiterals [
+	^ self readDefaultOption: #optionReadOnlyLiterals
+]
+
+{ #category : #'options - settings API' }
+CompilationContext class >> optionReadOnlyLiterals: aBoolean [
+	^ self writeDefaultOption: #optionReadOnlyLiterals value: aBoolean
+]
+
+{ #category : #options }
+CompilationContext class >> optionsDescription [
+	"Default options are held by DefaultOptions class variable.
+	 The description of default options here is only used as a 
+	 fall-back if DefaultOptions is not initialized (bootstrap, ...)"
+	
+	"each entry is fall-back default value (+/-), option name, description"
+	^ #(
+	(+ optionInlineIf 					'Inline ifTrue:, ifFalse:, ifTrue:ifFalse:, ifFalse:ifTrue: if specific conditions are met (See isInlineIf)')
+	(+ optionInlineIfNil 				'Inline ifNil:, ifNotNil:, ifNil:ifNotNil:, ifNotNil:ifNil: if specific conditions are met (See isInlineIfNil)')
+	(+ optionInlineAndOr 				'Inline and:, or: if specific conditions are met (See isInlineAndOr)')
+	(+ optionInlineWhile 				'Inline whileTrue, whileTrue:, whileFalse:, whileFalse if specific conditions are met (See isInlineWhile)')
+	(+ optionInlineToDo 				'Inline to:do:, to:by:do: if specific conditions are met (See isInlineToDo)')
+	(+ optionInlineCase 				'Inline caseOf:, caseOf:otherwise: if specific conditions are met (See isInlineCase)')
+	(- optionInlineTimesRepeat 		'Inline timesRepeat: if specific conditions are met (See isInlineTimesRepeat)')
+	(- optionInlineRepeat 			'Inline repeat if specific conditions are met (See isInlineRepeat)')
+	(- optionInlineNone 				'To turn off all inlining options. Overrides the others')
+	
+	(- optionReadOnlyLiterals 			'Compiler sets literals as read-only')
+	(- optionFullBlockClosure 			'Compiler compiles closure creation to use FullBlockClosure instead of BlockClosure')
+	(- optionLongIvarAccessBytecodes 	'Specific inst var accesses to Maybe context objects')
+	(+ optionOptimizeIR 					'Rewrite jumps in bytecode in a slightly more efficient way')
+	(- optionParseErrors 					'Parse syntactically wrong code')
+	)
+]
+
+{ #category : #'options - settings API' }
+CompilationContext class >> readDefaultOption: option [
+	^ self defaultOptions includes: option
 ]
 
 { #category : #removing }
 CompilationContext class >> removeDefaultTransformationPlugin: aPlugin [
 	self defaultTransformationPlugins remove: aPlugin
 ]
 
-{ #category : #accessing }
-CompilationContext class >> usesFullBlockClosure [
-	^ UsesFullBlockClosure ifNil: [ self initialize. UsesFullBlockClosure ]
-]
-
-{ #category : #accessing }
-CompilationContext class >> usesFullBlockClosure: usesFullBlock [
-	usesFullBlock ifTrue: [ 
-		(BytecodeBackend name endsWith: 'SistaV1') ifFalse: [ 
```

Comment:
Why we can't talk with objects using normal messages?
At least #isSista should be used. Or even better it should be:
```Smalltalk
BytecodeBackend confirmFullBlockClosureSupport
```
And I think it is better to use Warning instead of alert:
   self notify: 'warning'

Reply:
Ignore it . I am stupidly comment previous version